homeassistant/components/elmax/config_flow.py

```diff
@@ -40,6 +63,43 @@
     }
 )
 
+CHOOSE_MODE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ELMAX_MODE, default=CONF_ELMAX_MODE_CLOUD): SelectSelector(
+            SelectSelectorConfig(
+                options=[
+                    SelectOptionDict(
+                        value=CONF_ELMAX_MODE_CLOUD,
+                        label="Connect to Elmax Panel via Elmax Cloud APIs",
+                    ),
+                    SelectOptionDict(
+                        value=CONF_ELMAX_MODE_DIRECT,
+                        label="Connect to Elmax Panel via local/direct IP",
+                    ),
+                ],
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        )
+    }
+)
+
+DIRECT_SETUP_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ELMAX_MODE_DIRECT_HOST): str,
+        vol.Required(CONF_ELMAX_MODE_DIRECT_PORT, default=443): int,
+        vol.Required(CONF_ELMAX_MODE_DIRECT_SSL, default=True): bool,
+        vol.Required(CONF_ELMAX_MODE_DIRECT_FOLLOW_MDNS, default=True): bool,
```

Comment:
This seems like an unnecessary option, why would a user disable this?

Reply:
Sorry, which option are you referring to? 

About SSL, the user might want to disable it in case the exposed certificate from the panel is untrusted, expired or untrusted and therefore rely on plain-text http. In general not all Elmax panels support SSL and the majority are not equipped with Globally Signed SSL certificates.

About following MDNS, some power user might not want to rely on MDNS, instead they might want to use ARP binding/static DHCP lease  at router level instead of relying on mDNS discovery/following option.

Reply:
I meant both the SSL and mDNS options.

If we require user to enter host and port in direct mode, why is mDNS involved?

Reply:
Answered in another post.